vispy/color/colormap.py

```diff
@@ -115,45 +118,101 @@ def step(colors, x, controls=None):
     return colors[x_step, ...]
 
 
+# find the first 'controls' value that is smaller than 't'
+def find_color_index(controls=None, t=None, ncolors=None):
+    bn = np.nonzero(controls >= t)
+    j = bn[0][0]-1
+    j = np.clip(j, 0, ncolors-1)
+    return j
+
+
 # GLSL interpolation functions.
-def _glsl_mix(controls=None):
+def _glsl_mix(controls=None, colors=None, texture_map_data=None):
     """Generate a GLSL template function from a given interpolation patterns
-    and control points."""
+    and control points.
+
+    Parameters
+    ----------
+    colors : array-like, shape (n_colors, 4)
+        List of lists, tuples, ndarrays or ColorArrays.
+        The control colors used by the colormap.
+
+    controls : list
+        The list of control points for the given colors. It should be
+        an increasing list of floating-point number between 0.0 and 1.0.
+        The first control point must be 0.0. The last control point must be
+        1.0. The number of control points depends on the interpolation scheme.
+
+    texture_map_data : ndarray, shape(LUT_len, 4)
+        Numpy array of size of 1D texture lookup data
+        for luminance to RGBA conversion.
+        If texture_map_data is not None,
+        the RGBA data for the array is computed.
+
+    """
     assert (controls[0], controls[-1]) == (0., 1.)
     ncolors = len(controls)
     assert ncolors >= 2
-    if ncolors == 2:
-        s = "    return mix($color_0, $color_1, t);\n"
+    if texture_map_data is None:
+        s2 = ""
+        if ncolors == 2:
+            s = "    return mix($color_0, $color_1, t);\n"
+        else:
```

Comment:
Rather than putting the conditional code back in which I know @larsoner requested be removed, how about you update the tests to not need them? Or wouldn't it be possible to construct the `texture_map_data` from the provided colors and controls?